### ISSUE TITLE
Question on the installation of binny

### ISSUE THREAD

**Issue by B-1991-ing** [a-h-b/binny#26]
Dear binny support team,

I asked our HPC support team to install **binny** on the HPC. But our HPC team said that **it looks like binny is designed to be configured and run ONLY from a per-user installation, so it makes no sense to install it shared in Computerome.
So, you must install it yourself, preferably in your PROJECT structure.**

Is there any way that it can also be installed on the HPC cluster?

Best,

Bing

**Comment by B-1991-ing** [comment]
Dear Anna,

Thank you very much for your reply and information.

I am trying to install binny myself. I want to use **the newest version of conda, prokka, and snakemake**. But, error happened due to the path of the conda path. Do you think it should be solved by the HPC support team? I am also not sure which is the "**VARIABLE_CONFIG**".
<img width="384" alt="Screenshot 2022-06-09 at 23 32 02" src="https://user-images.githubusercontent.com/54441633/172948918-ecbf3593-1c29-4fd9-97c2-41b57665610b.png">

**Installation steps:**

**The 1st step**
git clone https://github.com/a-h-b/binny.git
cd binny

**The 2nd step**
my_conda_path="/services/tools/anaconda3/2021.11"
sed -i -e "s|conda_source: \"\"|conda_source: \"${my_conda_path}\"|g" config/config.*.yaml

**The 3rd step**
my_prokka_env="/services/tools/prokka/1.14.5" 
my_snakemake_env="/services/tools/snakemake/6.15.3" 
sed -i -e "s|prokka_env: \"\"|prokka_env: \"${my_prokka_env}\"|g" \
       -e "s|snakemake_env: \"\"|snakemake_env: \"${my_snakemake_env}\"|g" config/config.*.yaml

**The 4th step**
./binny -i config/config.init.yaml 

**Error message**
<img width="1354" alt="Screenshot 2022-06-09 at 23 03 20" src="https://user-images.githubusercontent.com/54441633/172948122-9c5b004a-01cd-403c-ad7a-c46e880f6a01.png">
<img width="1361" alt="Screenshot 2022-06-09 at 23 18 13" src="https://user-images.githubusercontent.com/54441633/172948136-f6f9389d-d54c-49a4-ba7f-56486a230271.png">
<img width="1356" alt="Screenshot 2022-06-09 at 23 18 31" src="https://user-images.githubusercontent.com/54441633/172948144-71248d43-7d3b-4db6-947a-beceb6153d3d.png">

The error is due to the conda path. When I search the "anaconda3/2021.11", it gave me "ANACONDA3_HOME", so I chose it.
<img width="1353" alt="Screenshot 2022-06-09 at 23 27 19" src="https://user-images.githubusercontent.com/54441633/172948485-f70d91d8-72bf-42a0-82e0-cd5515012e4a.png">

Best, 

Bing

**Comment by a-h-b** [comment]
Dear Bing -
I think you set a conda path that is not good - i.e. I doubt you are allowed to install things there and the installation does not find what it's looking for. Please set the conda_source in the config files to "". This will make snakemake and conda install the necessary environments in you new binny directory. 
```
my_conda_path='""'
sed -i -e "s|^conda_source: .*|conda_source: "${my_conda_path}"|g" config/config.*.yaml
```
Then try the installation again. And please let me know if that worked.
Best wishes -
Anna

**Comment by B-1991-ing** [comment]
Hi Anna,

I tried to install as below, still commands not found.

<img width="1359" alt="Screenshot 2022-06-10 at 10 20 11" src="https://user-images.githubusercontent.com/54441633/173024126-50f3c2e7-ea61-48d3-8438-0d6733102d75.png">

<img width="1359" alt="Screenshot 2022-06-10 at 10 20 34" src="https://user-images.githubusercontent.com/54441633/173024164-a9655167-ec38-4c1a-9b1c-9520659297f6.png">

Best,

Bing

**Comment by ohickl** [comment]
Hi Bing,

could you, as a sanity check, remove binny, clone the repo, just do the default setup, and run the test?
Also, the paths in `/services/tools/<tool>/<version>`, are these conda environments? binny assumes this, because snakemake works with them and tries to activate them as such.

Best

Oskar

**Comment by B-1991-ing** [comment]
Hi Oskar,

Thank you very much for your reply and suggestion.

I did as below. Still command not found.

<img width="1359" alt="Screenshot 2022-06-10 at 10 45 30" src="https://user-images.githubusercontent.com/54441633/173028152-aba0b370-485b-48be-a038-83edcc386db5.png">

<img width="1358" alt="Screenshot 2022-06-10 at 10 45 43" src="https://user-images.githubusercontent.com/54441633/173028176-81858647-8c0d-4c58-89c1-b84cd1613bce.png">

Best,

Bing

**Comment by ohickl** [comment]
Ah, it seems like you dont have conda in your `PATH`? Does it find anything if you e.g. type `which conda`? Maybe on your HPC environment you need to load it first, if you didnt install it personally, e.g with `module load anaconda3/2021.11` (guess based on one of your previous screen-shots)

**Comment by a-h-b** [comment]
if you usually need to start conda by loading a module, you can add the module load command to the VARIABLE_CONFIG. Let me know, if this is what you need to do -A

**Comment by B-1991-ing** [comment]
Hi Anna and Oskar,

I added **one(only conda**) or **three(conda, snakes make, prokka)** lines in my "VARIABLE_CONFIG" file - VARIABLE_CONFIG.txt.
[VARIABLE_CONFIG.txt](https://github.com/a-h-b/binny/files/8877554/VARIABLE_CONFIG.txt)

**LOADING_MODULES	module load anaconda3/2021.11**
**LOADING_MODULES	module load snakemake/6.15.3**
**LOADING_MODULES	module load prokka/1.14.5**

Error like below.
<img width="1195" alt="Screenshot 2022-06-10 at 11 19 24" src="https://user-images.githubusercontent.com/54441633/173034210-3bdb7f34-a9d3-4a8e-b20b-c7444342d357.png">

Best,

Bing

**Comment by a-h-b** [comment]
Hi Bing - sorry for the late reply. Can you please set SNAKEMAKE_VIA_CONDA to true in the VARIABLE_CONFIG? -A

**Comment by B-1991-ing** [comment]
Hi Anna,

My HPC admin finally installed binny on the HPC via miniconda for us this morning.

Thank you very much for your help.

Best,

Bing

**Comment by a-h-b** [comment]
Hi Bing - great thanks for letting us know. -A

**Comment by B-1991-ing** [comment]
Hi Anna,

Binny hasn't been installed on our HPC cluster yet, I saw the email from our HPC team wrongly.

So, I tried to install it in two ways as shown below.

**1. NAKEMAKE_VIA_CONDA true    LOADING_MODULES module load miniconda3/4.12.0** 
![image](https://user-images.githubusercontent.com/54441633/174443521-2c61de5d-19c6-47d4-bed4-6efd01fa65da.png)
<img width="1178" alt="Screenshot 2022-06-18 at 17 15 01" src="https://user-images.githubusercontent.com/54441633/174444855-8a2e1244-33ca-460e-b5ec-f3d4a1e27e6b.png">

**2. Try to install miniconda3 under my PROJECT directory, but no space left on the terminal**
<img width="751" alt="Screenshot 2022-06-18 at 17 11 19" src="https://user-images.githubusercontent.com/54441633/174444704-7819c47d-34fe-46cf-87b3-39f21c57edae.png">

I will continue emailing our HPC support team, to see if there is any way to directly install binny on the HPC cluster.

Best,

Bing
